salt/haproxy/haproxy.cfg.jinja

```diff
@@ -9,58 +9,77 @@ global
 
 defaults
         log     global
-        mode    tcp
-        option  tcplog
+        mode    http
+        option  redispatch
         option  dontlognull
-        timeout connect 5000
-        timeout client 120000
-        timeout server 120000
+        balance roundrobin
+        timeout connect      5s
+        timeout client       120s
+        timeout http-request 30s
+        timeout client-fin   30s
+        timeout server       120s
+        timeout tunnel       0
+        default-server inter 10s fall 2
 
 # Listen on the standard Kube-API Public port, 6443 by default, and proxy to the masters on
 # the Kube-API internal port, 6444 by default.
-listen kubernetes-master
-        bind {{ bind_ip }}:{{ pillar['api']['ssl_port'] }}
-        mode tcp
-        default-server inter 10s fall 2
-        balance roundrobin
-        option redispatch
+frontend kubernetes-master
+        bind {{ bind_ip }}:{{ pillar['api']['ssl_port'] }} ssl crt {{ pillar['ssl']['kube_apiserver_proxy_bundle'] }} ca-file {{ pillar['ssl']['ca_file'] }} verify optional
+        timeout client 0
+        http-request set-header X-Remote-User  %{+Q}[ssl_c_s_dn(cn)] if { ssl_c_used ssl_c_verify 0 }
+        http-request set-header X-Remote-Group %{+Q}[ssl_c_s_dn(o)] if { ssl_c_used ssl_c_verify 0 }
+        # If no certificate is passed, or if it's invalid, remove the auth headers
+        http-request del-header X-Remote-User  unless { ssl_c_used ssl_c_verify 0 }
+        http-request del-header X-Remote-Group unless { ssl_c_used ssl_c_verify 0 }
+        http-request del-header X-Remote-Extra unless { ssl_c_used ssl_c_verify 0 }
+        acl streaming_logs url_reg [?&]follow=true
+        use_backend no-timeout-backend if streaming_logs
+        acl interactive_session url_reg [?&]tty=true
+        use_backend no-timeout-backend if interactive_session
```

Comment:
(I'm not an haproxy expert, so) is this setting the `haproxy` as the SSL termination? Will the `haproxy`<->`master` traffic be in plain HTTP?

Reply:
It's TLS terminated.

Reply:
Ok, TLS, but I mean that the `haproxy`<->`master` communication will be in plain HTTP, right?

Reply:
No, that communication happens also over TLS.